security/remote_devices/docker-swarm/docker-stack-edgex.yml

```
@@ -0,0 +1,617 @@
+  secrets-volume:
+    driver: local
+    driver_opts:
+      o: bind
+      type: none
+      device: /tmp/edgex/secrets/
+  secrets-ca-volume:
+    driver: local
+    driver_opts:
+      o: bind
+      type: none
+      device: /tmp/edgex/secrets/ca/
+  edgex-consul:
+    driver: local
+    driver_opts:
+      o: bind
+      type: none
+      device: /tmp/edgex/secrets/edgex-consul/
+  edgex-vault:
+    driver: local
+    driver_opts:
+      o: bind
+      type: none
+      device: /tmp/edgex/secrets/edgex-vault/
+  edgex-kong:
+    driver: local
+    driver_opts:
+      o: bind
+      type: none
+      device: /tmp/edgex/secrets/edgex-kong/
+  edgex-security-proxy-setup:
+    driver: local
+    driver_opts:
+      o: bind
+      type: none
+      device: /tmp/edgex/secrets/edgex-security-proxy-setup/
+  edgex-redis:
+    driver: local
+    driver_opts:
+      o: bind
+      type: none
+      device: /tmp/edgex/secrets/edgex-redis/
+  docker-sock:
+    driver: local
+    driver_opts:
+      o: bind
+      type: none
+      device: /var/run/
+  edgex-support-notifications:
+    driver: local
+    driver_opts:
+      o: bind
+      type: none
+      device: /tmp/edgex/secrets/edgex-support-notifications/
+  edgex-core-metadata:
+    driver: local
+    driver_opts:
+      o: bind
+      type: none
+      device: /tmp/edgex/secrets/edgex-core-metadata/
+  edgex-core-data:
+    driver: local
+    driver_opts:
+      o: bind
+      type: none
+      device: /tmp/edgex/secrets/edgex-core-data/
+  edgex-support-scheduler:
+    driver: local
+    driver_opts:
+      o: bind
+      type: none
+      device: /tmp/edgex/secrets/edgex-support-scheduler/
+  edgex-core-command:
+    driver: local
+    driver_opts:
+      o: bind
+      type: none
+      device: /tmp/edgex/secrets/edgex-core-command/
+
+services:
+  consul:
+    image: edgexfoundry/docker-edgex-consul:1.2.0
+    ports:
+      - "8400:8400"
+      - "8500:8500"
+    hostname: edgex-core-consul
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-core-consul
+    volumes:
+      - consul-config:/consul/config:z
+      - consul-data:/consul/data:z
+      - consul-scripts:/consul/scripts:z
+      - secrets-ca-volume:/tmp/edgex/secrets/ca:ro,z
+      - edgex-consul:/tmp/edgex/secrets/edgex-consul:ro,z
+      - edgex-vault:/tmp/edgex/secrets/edgex-vault:ro,z
+      - edgex-kong:/tmp/edgex/secrets/edgex-kong:ro,z
+    environment:
+      - "SECRETSTORE_SETUP_DONE_FLAG=/tmp/edgex/secrets/edgex-consul/.secretstore-setup-done"
+      - EDGEX_DB=redis
+      - EDGEX_SECURE=true
+      - CONSUL_BIND_INTERFACE=eth0
+    depends_on:
+      - security-secrets-setup
+    deploy: 
+      placement: 
+        constraints: 
+          - node.hostname != worker-node 
+
+  vault:
+    image: vault:1.3.1
+    hostname: edgex-vault
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-vault
+    ports:
+      - "8200:8200"
```

```suggestion
      - target: 8200
        published: 8200
        protocol: tcp
        mode: host
```

Above change needed for dnsrr change below